parcels/particle.py

```diff
@@ -224,12 +234,12 @@ def Kernel(self, pyfunc):
         return Kernel(self.grid, self.ptype, pyfunc)
 
     def ParticleFile(self, *args, **kwargs):
-        return ParticleFile(*args, particleset=self, **kwargs)
+        return ParticleFile(*args, particleset=self, user_vars=self.ptype.user_vars, **kwargs)
 
 
 class ParticleFile(object):
 
-    def __init__(self, name, particleset, initial_dump=True):
+    def __init__(self, name, particleset, initial_dump=True, user_vars=None):
```

Comment:
Ah yes, that's much cleaner- I'd forgotten i'd added that attribute to
ParticleType. Cool, thanks Michael.

On 8 March 2016 at 07:41, Michael Lange <notifications@github.com> wrote:

> In parcels/particle.py
> <https://github.com/OceanPARCELS/parcels/pull/43#discussion_r55321701>:
>
> >
> >
> >  class ParticleFile(object):
> >
> > -    def __init__(self, name, particleset, initial_dump=True):
> > +    def __init__(self, name, particleset, initial_dump=True, user_vars=None):
>
> The particleset has an associated ParticleType ptype attribute, which
> contains the user_vars. We could just infer it from there instead of
> adding to the constructor, no?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OceanPARCELS/parcels/pull/43/files#r55321701>.
>